File: src/test/java/CustomerDataTests.java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import se.gewalli.commands.Command;
import se.gewalli.data.EntityNotFound;
import se.gewalli.data.InMemoryRepository;
import se.gewalli.data.Repository;
import xmlimport.GetCommands;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CustomerDataTests {
    Repository repository;
    @BeforeEach
    public void beforeEach() throws EntityNotFound {
        repository=new InMemoryRepository();
        var getCommands=new GetCommands();
        for (Command command : getCommands.Get()) {
            command.run(repository);
        }
    }
    @Test
    public void canGetCustomerById() throws EntityNotFound {
        assertEquals("Steve" , repository.getCustomer(1).firstname);
    }
    @Test
    public void canGetProductById() throws EntityNotFound {
        assertEquals("Yo-yo" , repository.getProduct(1).name);
    }
    @Test
    public void orderContainsProduct() throws EntityNotFound {
        assertEquals(3 , repository.getOrder(1).products.size());
    }
}

File: src/test/java/CustomerDataTests.kt
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import se.gewalli.data.EntityNotFound
import se.gewalli.data.InMemoryRepository
import se.gewalli.data.Repository
import xmlimport.GetCommands

class CustomerDataTests {
    var repository: Repository? = null
    @BeforeEach
    @Throws(EntityNotFound::class)
    fun beforeEach() {
        repository = InMemoryRepository()
        val getCommands = GetCommands()
        for (command in getCommands.Get()) {
            command.run(repository!!)
        }
    }

    @Test
    @Throws(EntityNotFound::class)
    fun canGetCustomerById() {
        Assertions.assertEquals("Steve", repository!!.getCustomer(1).firstname)
    }

    @Test
    @Throws(EntityNotFound::class)
    fun canGetProductById() {
        Assertions.assertEquals("Yo-yo", repository!!.getProduct(1).name)
    }

    @Test
    @Throws(EntityNotFound::class)
    fun orderContainsProduct() {
        Assertions.assertEquals(3, repository!!.getOrder(1).products.size)
    }
}